### PR TITLE
Add `InteropQubitManager` to use in Cirq/Bloqs interop code.

### PR DESCRIPTION
**Pull request by tanujkhattar**
Fixes https://github.com/quantumlib/Qualtran/issues/384

### CODE REVIEW

File: qualtran/cirq_interop/_interop_qubit_manager_test.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import cirq
+import pytest
+
+from qualtran.cirq_interop._interop_qubit_manager import InteropQubitManager
+
+
+def test_interop_qubit_manager():
```
Comment (fdmalone) [inline qualtran/cirq_interop/_interop_qubit_manager_test.py:20]: Is it worth adding a gate which uses this (like the and / and^ example from the issue?)

Comment (fdmalone) [reply qualtran/cirq_interop/_interop_qubit_manager_test.py:20]: Adding as a unit test I mean

**Review by fdmalone — APPROVED** [review]
Approved % suggestion for a test.

File: qualtran/cirq_interop/_interop_qubit_manager_test.py (hunk 1 above)
Comment (tanujkhattar) [reply qualtran/cirq_interop/_interop_qubit_manager_test.py:20]: added a test. Merging now.

File: qualtran/cirq_interop/_interop_qubit_manager.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#  limitations under the License.
+
+"""Qubit Manager to use when converting Cirq gates to/from Bloqs."""
+from typing import Iterable
+
+import cirq
+
+
+class InteropQubitManager(cirq.ops.SimpleQubitManager):
+    """Qubit Manager to use to facilitate interop of Cirq gates and Bloqs."""
```
Comment (mpharrigan) [inline qualtran/cirq_interop/_interop_qubit_manager.py:22]: what's the difference between this and simplequbitmanager?

**Comment by mpharrigan** [comment]
can you please describe the root cause of the problem in the linked issue and why the fix works

File: qualtran/cirq_interop/_interop_qubit_manager.py (hunk 2 above)
Comment (tanujkhattar) [reply qualtran/cirq_interop/_interop_qubit_manager.py:22]: You can provide it externally allocated qubits and ask it to "manage" them. Essentially the `manage_qubits` method that was added.

**Comment by tanujkhattar** [comment]
@mpharrigan  added an explanation in https://github.com/quantumlib/Qualtran/issues/384#issuecomment-1754081427

**Pull request merged**